book/src/04_4_interrupts.md

```diff
@@ -11,32 +11,36 @@ TODO why are some functions called with some(...), esp!(...) or "normal"?
 
 ## Tasks
 
-1. Configure the button (GPIO 9) with a c struct [`gpio_config_t`](https://esp-rs.github.io/esp-idf-sys/esp_idf_sys/struct.gpio_config_t.html)the following settings:
+✅ Configure the button GPIO 9 with a c struct [`gpio_config_t`](https://esp-rs.github.io/esp-idf-sys/esp_idf_sys/struct.gpio_config_t.html)the following settings (1. in the exercise/src/main.rs):
     - input mode
     - pull up
     - interrupt on positive edge
   
-  TODO add the variable names for the settings
+    
+    
+### Hints!
 
-2. Write the configuration into the register with [`unsafe extern "C" fn gpio_config`](https://esp-rs.github.io/esp-idf-sys/esp_idf_sys/fn.gpio_config.html). This needs to happen in the unsafe block. To make these FFI calls we can use the macro `esp!($Cfunktion)`.
+- Use bit shift to set a `1` in the right position with the constant GPIO_NUM.
+- The fields `pull_up_en` and `pull_down_en` can be set with `true/false.into()`, or just `0/1`.
+- The other two constants you need to set to true have been imported by `esp_idf_sys`, so use those or just 1. 
```

Comment:
I have added more explanation for this in my PR so people can work more independent.